index.js

```diff
@@ -397,9 +368,7 @@ Router.prototype.find = function find (method, path, version) {
     var previousPath = path
     // found the route
     if (pathLen === 0 || path === prefix) {
-      var handle = version === undefined
-        ? currentNode.handler
-        : currentNode.getVersionHandler(version)
+      var handle = currentNode.getMatchingHandler(derivedConstraints)
```

Comment:
you lose throughput here, this function call was not in the hot path before.

Reply:
Indeed, but weirdly, on @AyoubElk 's branch he also switched this to making a function call every time and had the static routing performance inline with `master`. It also seems like one more monomorphic function call shouldn't cause a 30% regression? I also tried commenting out the function call and returning just `.handlers[0]` to mimic the old behaviour just to test performance, and performance didn't really recover much. So, I feel like I'm missing something :/

Reply:
Updated to not make this call unless there are routes with constraints